src/aatoolbox/datasources/codab/codab.py

```diff
@@ -49,50 +45,101 @@ def download(
 
         Examples
         --------
-        >>> from aatoolbox.datasources.codab.codab import CodAB
+        >>> from aatoolbox import create_country_config, CodAB
         >>> # Download COD administrative boundaries for Nepal
-        >>> codab = CodAB("npl")
-        >>> npl_cod_shapefile = codab.download(
-        ...    hdx_address="administrative-bounadries-of-nepal",
-        ...    hdx_dataset_name="npl_admbnda_nd_20201117_SHP.zip")
+        >>> country_config = create_country_config(iso3="npl")
+        >>> codab = CodAB(country_config=country_config)
+        >>> npl_cod_shapefile = codab.download()
         """
         return self._download(
             filepath=self._raw_filepath,
-            hdx_address=hdx_address,
-            hdx_dataset_name=hdx_dataset_name,
+            hdx_address=self._country_config.codab.hdx_address,
+            hdx_dataset_name=self._country_config.codab.hdx_dataset_name,
             clobber=clobber,
         )
 
     @staticmethod
```

Comment:
Yes it's static in order to use the decorator, but indeed it could be made into a module function. To be honest that is a question I often have: When to use a static method vs a module function? Do you have any advice for how to decide?

As for the decorator, I actually wanted to write it to be used on instance method, but I ran into some issues, I forget exactly why. I've made an issue [here](https://github.com/OCHA-DAP/pa-aa-toolbox/issues/62) to either fix the decorator or the methods that it's applied to (it's also done this way elsewhere in the code).